src/lib/storage/index/base_index.cpp

```diff
@@ -1,10 +1,29 @@
 #include "base_index.hpp"
 
+#include <limits>
 #include <memory>
 #include <vector>
 
+#include "storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp"
+#include "storage/index/group_key/composite_group_key_index.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+
 namespace opossum {
 
+uintptr_t BaseIndex::predict_memory_consumption(ColumnIndexType type, ChunkOffset row_count, ChunkOffset value_count,
+                                                uint32_t value_bytes) {
+  switch (type) {
+    case ColumnIndexType::GroupKey:
+      return GroupKeyIndex::predict_memory_consumption(row_count, value_count, value_bytes);
+    case ColumnIndexType::CompositeGroupKey:
+      return CompositeGroupKeyIndex::predict_memory_consumption(row_count, value_count, value_bytes);
+    case ColumnIndexType::AdaptiveRadixTree:
+      return AdaptiveRadixTreeIndex::predict_memory_consumption(row_count, value_count, value_bytes);
```

Comment:
Since this is not implemented and you handle „not implemented“ in the default case, you could remove that method from the ART.

Reply:
True. We would opt to leave it in for consistency since we added the purely virtual `size_t BaseIndex::_memory_consumption()` method which needs to be implemented. If you would like us to take it out, however, we could do it.

Reply:
No strong opinion about this.